### DISCUSSION TITLE
Compile Custom Build for ESP32-S2 Mini

### DISCUSSION THREAD

**Discussion by KaiBuesing** [arendst/Tasmota#18316] [Q&A]
I would like to create a custom build for the ESP32-S2 Mini. Reason - I want to define my WLAN data and need the display driver for the SH1106. In platformio_override.ini I selected "tasmota32s2" and activated "board = esp32" at [env:tasmota32_base]. WLAN Settings I have entered user_config_override.h. The build is also generated without errors, but does not work on my ESP32-S2. Any idea what I have forgotten ?
<img width="866" alt="image" src="https://user-images.githubusercontent.com/113059963/229290771-6665a15e-5936-4116-a4fa-cfd887efe65c.png">
<img width="645" alt="image" src="https://user-images.githubusercontent.com/113059963/229290795-b86db488-9c0d-46e2-bf63-f081248e444a.png">
<img width="760" alt="image" src="https://user-images.githubusercontent.com/113059963/229290822-a9070f8b-6dc2-4ee7-ac1d-043b92c8c8b9.png">

**Comment by sfromis** [comment]
You should not change that "board" setting, this is likely where you got into trouble. Merely select `tasmota32s2` if you want no serial (USB CDC) access via USB, or probably `tasmota32s2cdc` if you want to have this available too.

And you should flash the `.factory.bin` file created by the build, unless you are only upgrading an already working Tasmota setup.

**Comment by Jason2866** [comment]
There is no other change needed, just uncomment the build variant env you want to have compiled.
Thats it. All other changes needs to be done in `user_config_override.h`

**Comment by KaiBuesing** [comment]
Thanks for the quick responses! My problem is solved :-)
